src/plugins/home/public/application/components/add_data/add_data.tsx

```diff
@@ -38,6 +38,8 @@ interface Props {
 export const AddData: FC<Props> = ({ addBasePath, application, isDarkMode, isCloudEnabled }) => {
   const { trackUiMetric, guidedOnboardingService } = getServices();
   const canAccessIntegrations = application.capabilities.navLinks.integrations;
+  const onlyReadAccess = !application.capabilities.rulesSettings.writeFlappingSettingsUI;
```

Comment:
I don't have a lot of experience with capabilities, but I don't think this is a good flag for this readonly state. `rulesSettings.writeFlappingSettingsUI` seem to be a alerting framework 
specific capabilities and we shouldn't use it outside of it. 


I think this is the right direction, but I don't know if there is an existing capability that could work for us here (need to look for it) otherwise we might have to register a new one

Reply:
For the the sampel data, I think `application.capabilities.indexPatterns.save` would be a good capability. To create sample data you'd have to be able to create a (dataview) so it would at least be a required condition. 

for the `fileDataViz` not sure

Reply:
Also the screenshot from the issue https://github.com/elastic/kibana/issues/138547 is pointing out to the different component

Reply:
In the comment below the body of the issue there was this suggestion
https://github.com/elastic/kibana/issues/138547#issuecomment-1674967169, which made sense, so I followed it. 
Do you think it is less beneficial to disable the whole accordeon?

Reply:
I was thinking a bit more about the issue and did some testing. I think the issue is not that simple and we will need to find some middleground between the effort and the result. Some investigation is needed, here are some pointers: 

- In general, just disabling the buttons is not a good UX. It either need to have a reason (for example in a tooltip) explaining why it is disable or simply hidden completely. 
- Hidding or disabling the buttons in `add_data` component is not enough, because there are other ways to navigate to `app/home#/tutorial_directory/sampleData` page . Also maybe we shouldn't disable those buttons in `add_data` component at all , because there is an external link from that page to a demo environment that isn't restricted by permissions
- `application.capabilities.savedObjectsManagement.edit` is a good guess for checking if the user is able to add sample data, but it might give false positives and negatives. Ideally we should investigate exact permissions that are needed for adding sample data and then decide what to do with them. This can be done by reviewing the code, analysing internal errors and trial and error. 
- I am not sure if it is correct to treat sample data and import together. They likely need different inner permissions. I was exploring the upload functionality and found that there is a more specific check inside https://github.com/elastic/kibana/blob/3177b037d7f8b52214b025a5f1128bd991efc591/x-pack/plugins/data_visualizer/public/application/file_data_visualizer/components/file_data_visualizer_view/file_data_visualizer_view.js#L72-L75 . It might be that reviewing it might help to better understand how permissions work and what how to better fix it.
- Another problem is that when you don't have enough permissions and try to add sample data we show "Internal server error" If we'd start from showing the proper "Permissions" error that we get from elasticsearch, that would be better for a user and for us to debug and improve the issue

Reply:
Thank you for your detailed review.
I will close this PR for now, since it doesn't fall into the "papercut" category.

Might reopen in the future.